syntax = "proto3";
package distinct;
import "auth.proto";
service InventoryService {
  rpc Post(PostRequest) returns (StatusReply);
  rpc Get(GetRequest) returns (GetResponse);
}

message PostRequest {
  string token = 1;
  string type  = 2;
  string sku   = 3;
  int32 qty    = 4;
}

message GetRequest {
  string token = 1;
  string type  = 2;
  string sku   = 3;
}

message Item {
  string sku = 1;
  string name = 2;
  int32 stock = 3;
}

message GetResponse {
  string status = 1;
  repeated Item items = 2;
  string message = 3;
}


